lib/task.js

```diff
@@ -39,20 +42,23 @@ const task = (dispatch) => (props, operationCreator, instanceUid = null) => {
   // First thing we need to do is create a `uid` for this task
   // `hashes` is an object with input, output, params keys
   // The `uid` is a hash of the `input`, `output`, and `params` hashes
-  const { uid, hashes } = generateUid(props)
+  let { uid, hashes } = generateUid(props)
```

Comment:
This is cool and in fact it is more clear to distinguish between `uid` and `newUid` because they are added to different objects. `newUid` is used when task is in fact created and `uid` is used out of the scope of the pipeline so to the definition of the task itself before having `context`.